bugsnag-plugin-android-ndk/src/main/jni/bugsnag_ndk.c

```diff
@@ -400,168 +421,196 @@ JNIEXPORT void JNICALL
-    (*env)->ReleaseStringUTFChars(env, new_value, value);
+    bsg_safe_release_string_utf_chars(env, new_value, value);
   }
 }
 
 JNIEXPORT void JNICALL Java_com_bugsnag_android_ndk_NativeBridge_updateUserName(
     JNIEnv *env, jobject _this, jstring new_value) {
-  if (bsg_global_env == NULL)
+  if (bsg_global_env == NULL) {
     return;
-  char *value = new_value == NULL
-                    ? NULL
-                    : (char *)(*env)->GetStringUTFChars(env, new_value, 0);
+  }
+  char *value = (char *)bsg_safe_get_string_utf_chars(env, new_value);
+  if (value == NULL) {
+    return;
+  }
   bsg_request_env_write_lock();
   bugsnag_event event = bsg_global_env->next_event;
   bugsnag_event_set_user(&bsg_global_env->next_event, event.user.id,
                          event.user.email, value);
   bsg_release_env_write_lock();
   if (new_value != NULL) {
-    (*env)->ReleaseStringUTFChars(env, new_value, value);
+    bsg_safe_release_string_utf_chars(env, new_value, value);
   }
 }
 
 JNIEXPORT void JNICALL
 Java_com_bugsnag_android_ndk_NativeBridge_updateUserEmail(JNIEnv *env,
                                                           jobject _this,
                                                           jstring new_value) {
-  if (bsg_global_env == NULL)
+  if (bsg_global_env == NULL) {
+    return;
+  }
+  char *value = (char *)bsg_safe_get_string_utf_chars(env, new_value);
+  if (value == NULL) {
     return;
-  char *value = new_value == NULL
-                    ? NULL
-                    : (char *)(*env)->GetStringUTFChars(env, new_value, 0);
+  }
   bsg_request_env_write_lock();
   bugsnag_event event = bsg_global_env->next_event;
   bugsnag_event_set_user(&bsg_global_env->next_event, event.user.id, value,
                          event.user.name);
   bsg_release_env_write_lock();
   if (new_value != NULL) {
-    (*env)->ReleaseStringUTFChars(env, new_value, value);
+    bsg_safe_release_string_utf_chars(env, new_value, value);
   }
 }
 
 JNIEXPORT void JNICALL
 Java_com_bugsnag_android_ndk_NativeBridge_addMetadataString(
     JNIEnv *env, jobject _this, jstring tab_, jstring key_, jstring value_) {
-  if (bsg_global_env == NULL)
+  if (bsg_global_env == NULL) {
     return;
-  char *tab = (char *)(*env)->GetStringUTFChars(env, tab_, 0);
-  char *key = (char *)(*env)->GetStringUTFChars(env, key_, 0);
-  char *value = (char *)(*env)->GetStringUTFChars(env, value_, 0);
-  bsg_request_env_write_lock();
-  bugsnag_event_add_metadata_string(&bsg_global_env->next_event, tab, key,
-                                    value);
-  bsg_release_env_write_lock();
-  (*env)->ReleaseStringUTFChars(env, tab_, tab);
-  (*env)->ReleaseStringUTFChars(env, key_, key);
-  (*env)->ReleaseStringUTFChars(env, value_, value);
+  }
+  char *tab = (char *)bsg_safe_get_string_utf_chars(env, tab_);
+  char *key = (char *)bsg_safe_get_string_utf_chars(env, key_);
+  char *value = (char *)bsg_safe_get_string_utf_chars(env, value_);
```

Comment:
We should definitely eliminate these casts away from const in a later PR...

Reply:
Raised PLAT-6035 to do this. The problem is that a lot of our public API specifies `char *` rather than `const char *` - I'm assuming that altering this would count as a breaking change. I can create a separate ticket to address the internal usages where possible.